### ISSUE TITLE
Missing File - WITH_SIFTGPU Off

### ISSUE THREAD

**Issue by CWallbridge** [strands-project/v4r#70]
While trying to build this project while WITH_SIFTGPU is set to Off I am getting the following error:

`<v4r directory>/modules/features/src/sift_local_estimator.cpp:11:54: fatal error: v4r/features/opencv_sift_local_estimator.h: No such file or directory
compilation terminated.
modules/features/CMakeFiles/v4r_features.dir/build.make:206: recipe for target 'modules/features/CMakeFiles/v4r_features.dir/src/sift_local_estimator.cpp.o' failed
make[2]: *** [modules/features/CMakeFiles/v4r_features.dir/src/sift_local_estimator.cpp.o] Error 1
CMakeFiles/Makefile2:1420: recipe for target 'modules/features/CMakeFiles/v4r_features.dir/all' failed
make[1]: *** [modules/features/CMakeFiles/v4r_features.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2`

Taking a quick look in sift_local_estimator.cpp you can see that the program is looking for opencv_sift_local_estimator.h based on HAVE_SIFTGPU:

`#ifdef HAVE_SIFTGPU
#include <v4r/features/sift_local_estimator.h>
#include <GL/glut.h>
#else
#include <v4r/features/opencv_sift_local_estimator.h>
#endif`

Looking in the appropriate folder opencv_sift_local_estimator.h is indeed missing. It is also not in this repository. I decided to take a quick look in sift_local_estimator.h where there is also a decision based on HAVE_SIFTGPU:

`#ifdef HAVE_SIFTGPU
#include <SiftGPU/SiftGPU.h>
#else
#include <opencv2/nonfree/features2d.hpp> // requires OpenCV non-free module
#endif`

So it might be that it has been updated to use sift_local_estimator.h in both cases at some point? I will see if modifying the code helps and see if I run into further issues.

**Comment by mzillich** [comment]
Why would you want to compile without SiftGPU? We do have GPUs in the robots.

**Comment by CWallbridge** [comment]
I am currently trying to install on a machine without a working GPU to explore this project until my new one arrives. Is installation without SIFTGPU no longer supported?

**Comment by mzillich** [comment]
Ah ok. In this case you have to install opencv from source, includng the nonfree parts (wich are not in the ubuntu standard opencv package)

**Comment by CWallbridge** [comment]
Well, really the issue is that, if `HAVE_SIFTGPU` is false, they we are supposed to include a header that does not exist:
 
``` cpp
#ifdef HAVE_SIFTGPU
#include <v4r/features/sift_local_estimator.h>
#include <GL/glut.h>
#else
#include <v4r/features/opencv_sift_local_estimator.h>
#endif
```

**Comment by mzillich** [comment]
Ha, that is odd. I do have that file in my version of the repo. You installed the debian v4r package?

**Comment by CWallbridge** [comment]
No, as I am actually trying this in Ubuntu 16.04, which you don't have a debian package for yet, so I am going from this source repository.
